### PR TITLE
Improve State Navigation using Right/Left Arrow Buttons

### PR DESCRIPTION
**Pull request by muddana-satish**
Right Arrow selects next available state.
Left Arrow selects previous available state.
shift+Right/shift+Left selects entire area until next/prev available state.

Signed-off-by: muddana-satish <satish.muddana@ericsson.com>

### CODE REVIEW

**Comment by tahini** [comment]
> Right+shift and Left+shift works correctly.
> 
Please define "correctly"?

File: timeline-chart/src/layer/time-graph-chart-cursors.ts [hunk 1]
```diff
@@ -179,11 +169,26 @@ export class TimeGraphChartCursors extends TimeGraphChartLayer {
 
     centerCursor() {
         if (this.unitController.selectionRange) {
-            const cursorPosition = this.unitController.selectionRange.start;
+            const cursorPosition = this.unitController.selectionRange.end;
             const halfViewRangeLength = this.unitController.viewRangeLength / 2;
+            let startViewRange = cursorPosition - halfViewRangeLength;
+            let endViewRange = cursorPosition + halfViewRangeLength;
+
+            let delta = 0;
```
Comment (tahini) [inline timeline-chart/src/layer/time-graph-chart-cursors.ts:177]: You don't need the delta variable here, you can just do the operations directly in each of the blocks

**Review by tahini — APPROVED** [review]
Works well! Thanks!

I have a minor comment on the code. Also, when we need performance improvement, a binary search could be used instead of a find, but it's fine for now.

File: timeline-chart/src/layer/time-graph-chart-cursors.ts (hunk 1 above)
Comment (muddana-satish) [reply timeline-chart/src/layer/time-graph-chart-cursors.ts]: updated. Thanks.

**Comment by muddana-satish** [comment]
> > Right+shift and Left+shift works correctly.
> 
> Please define "correctly"?

Updated the commit message.

**Comment by tahini** [comment]
Here's some issue I noticed while testing with the example timeline. Look at the following screencast, where I do left/right with one of the row with 2 states. I just click right till the end, then come back left till the end, again and again. Notice that the zoom level seems to change, or at least the state gets bigger when I come back to it later, until at some point they don't appear anymore, I just go from far right to far left and back.

leftRightTimeline

**Comment by muddana-satish** [comment]
@tahini  Thank you for catching this. I broke this while trying to address your suggestion on removing 'delta'. I fixed it. Thanks.

File: timeline-chart/src/layer/time-graph-chart.ts [hunk 2]
```diff
@@ -207,6 +210,7 @@ export class TimeGraphChart extends TimeGraphChartLayer {
     }
 
     protected async maybeFetchNewData(update?: boolean) {
+        console.log("Jaffa");
```
Comment (tahini) [inline timeline-chart/src/layer/time-graph-chart.ts:213]: translate? :p

File: timeline-chart/src/layer/time-graph-chart.ts [hunk 3]
```diff
@@ -269,6 +276,7 @@ export class TimeGraphChart extends TimeGraphChartLayer {
     }
 
     protected addRow(row: TimelineChart.TimeGraphRowModel, height: number, rowIndex: number) {
+        console.log("Po ra Daffa Nuvvu cheppedhi nenenti vinedhi");
```
Comment (tahini) [inline timeline-chart/src/layer/time-graph-chart.ts:279]: 2 possible translations: "Po ra daffa I hear what you say" or "Poe's what i have to say you're wrong, vinedi". Sounds right?

**Review by tahini — COMMENTED** [review]
Works well, but some debug artefacts remain...

File: timeline-chart/src/layer/time-graph-chart.ts (hunk 3 above)
Comment (muddana-satish) [reply timeline-chart/src/layer/time-graph-chart.ts]: Sorry for the mess. Cleared it. :/

**Review by tahini — APPROVED** [review]
Thanks!

**Review by bhufmann — APPROVED** [review]
Looks good to me! Thanks

File: timeline-chart/src/layer/time-graph-chart.ts (hunk 3 above)
Comment (mad-season) [reply timeline-chart/src/layer/time-graph-chart.ts]: Ha. Satish dropped some Telugu on us. @muddana-satish do at least tell us what it means. The suspense is killing me

Comment (muddana-satish) [reply timeline-chart/src/layer/time-graph-chart.ts]: It means: _Go Away! Why should I listen to you._ 
That's me interacting with JS engine.

**Comment by bhufmann** [comment]
Thanks for the contribution!

**Pull request merged**